ion/textwriter.go

```diff
@@ -393,8 +393,15 @@ func (w *textWriter) writeAnnotations() error {
 	as := w.annotations
 	w.annotations = nil
 
+	var text string
 	for _, a := range as {
-		if err := writeSymbol(a, w.out); err != nil {
+		if a.Text != nil {
+			text = *a.Text
+		} else {
+			text = "$0"
```

Comment:
Similar thoughts here about quietly writing out a `$0`.

Reply:
text is now set to `$id` unless id is -1 (then it returns an error).